mobile/app/(app)/change-password.tsx

```diff
@@ -0,0 +1,189 @@
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      currentPassword: "Votemonitor1*",
+      newPassword: "Votemonitor1*",
+      confirmPassword: "Votemonitor1*",
+    },
+  });
+  const { errors } = formState;
+
+  // Submit handler - change password
+  const {
+    mutate: updatePassword,
+    isError: credentialsError,
+    isSuccess: succesfullyChanged,
+  } = useChangePasswordMutation();
+
+  const onSubmit = async (data: FormData) => {
+    const payload: ChangePasswordPayload = {
+      password: data.currentPassword.trim(),
+      newPassword: data.newPassword.trim(),
+      confirmNewPassword: data.confirmPassword.trim(),
+    };
+    updatePassword(payload);
+  };
+
+  // Render either form or confirmation screen
+  return succesfullyChanged ? (
+    <ChangePasswordConfirmation />
+  ) : (
+    <Screen
+      preset="auto"
+      backgroundColor="white"
+      ScrollViewProps={{
+        bounces: false,
+      }}
+      contentContainerStyle={$containerStyle}
+    >
+      <Header
+        title={t("header.title")}
+        titleColor="white"
+        barStyle="light-content"
+        leftIcon={<Icon icon="chevronLeft" color="white" />}
+        onLeftPress={() => router.back()}
+      />
+
+      <YStack gap="$xl" paddingHorizontal={16} paddingVertical={40}>
+        <PasswordInput
+          formKey="currentPassword"
+          control={control}
+          helper={""}
+          error={
+            (errors?.currentPassword?.message?.toString() ?? "") ||
+            (credentialsError ? t("form.current_password.credentials_error") : "")
+          }
+          label={t("form.current_password.label")}
+          placeholder={t("form.current_password.placeholder")}
+          hasError={!!errors?.currentPassword || credentialsError}
+          name="currentPassword"
+        />
+
+        <PasswordInput
+          formKey="newPassword"
+          control={control}
+          helper={t("form.new_password.helper")}
+          error={errors?.newPassword?.message?.toString() ?? ""}
+          label={t("form.new_password.label")}
+          placeholder={t("form.new_password.placeholder")}
+          hasError={!!errors?.newPassword}
+          name="newPassword"
+        />
+
+        <PasswordInput
+          formKey="confirmPassword"
+          control={control}
+          helper={t("form.confirm_password.helper")}
+          error={errors?.confirmPassword?.message?.toString() ?? ""}
+          label={t("form.confirm_password.label")}
+          placeholder={t("form.confirm_password.placeholder")}
+          hasError={!!errors?.confirmPassword}
+          name="confirmPassword"
+        />
+      </YStack>
+
+      <Card width="100%" paddingBottom={16 + insets.bottom} marginTop="auto">
+        <Button onPress={handleSubmit(onSubmit)}>{t("form.actions.save_password")}</Button>
+      </Card>
+    </Screen>
+  );
+};
+
+interface PasswordInputProps {
+  control: Control<FormData, any>;
+  helper: string;
```

Comment:
this should be optional and only display it if we receive it

Reply:
done